File: 2021_spring_winter_algorithm_study_basic/WEEK01_TimeComplexity_BruteForce_Backtracking/2798.py
"""
input :
10 500
93 181 245 214 315 36 185 138 216 295
"""

import sys
import itertools

sys.stdin = open("input.txt", "r")
input = sys.stdin.readline

N, target_num = map(int, input().split())
num_list = list(map(int, input().split()))

ans = -1
for a, b, c in list(itertools.combinations(num_list, 3)):
    total = a + b + c
    if ans < total <= target_num:
        ans = total
print(ans)

File: 2021_spring_winter_algorithm_study_basic/WEEK01_TimeComplexity_BruteForce_Backtracking/2798.cpp
//
// Created by 김주호 on 2021/03/25.
//

#include <bits/stdc++.h>

using namespace std;
#define endl '\n'
#define FOR(i, n) for (int i= 0; i<n; i++)

int N, M, a[101], ans;

int main() {
    freopen("../input.txt", "r", stdin);
    cin >> N >> M;

    FOR(i, N) cin >> a[i];

    FOR(i, N) FOR(j, N) FOR(k, N)
        if (i != j && j != k && k != i) {
            int total = a[i] + a[j] + a[k];
            if (total <= M && ans < total) ans = total;
        }

    cout << ans << endl;
    return 0;
}